accounts/abi/abi.go

```diff
@@ -228,17 +228,36 @@ func (abi ABI) unmarshal(v interface{}, name string, output []byte) error {
 		if err != nil {
 			return err
 		}
-		reflectValue := reflect.ValueOf(marshalledValue)
-		if typ.AssignableTo(reflectValue.Type()) {
-			value.Set(reflectValue)
-		} else {
-			return fmt.Errorf("abi: cannot unmarshal %v in to %v", reflectValue.Type(), value.Type())
+		if err := set(value, reflect.ValueOf(marshalledValue), method.Outputs[0]); err != nil {
+			return err
 		}
 	}
 
 	return nil
 }
 
+func set(dst, src reflect.Value, output Argument) error {
+	dstType := dst.Type()
+	srcType := src.Type()
+
+	switch {
+	case dstType.AssignableTo(src.Type()):
+		dst.Set(src)
+	case dstType.Kind() == reflect.Array && srcType.Kind() == reflect.Slice:
```

Comment:
If `srcType` == slice, that can only happen in case of a byte slice right? Otherwise this if catches too broadly. If indeed only byte slices are possible, could you add a small comment mentioning this. It would help with code clarity.